core/src/test/java/org/elasticsearch/index/query/ExistsQueryBuilderTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermRangeQuery;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.elasticsearch.index.mapper.FieldMapper;
+import org.elasticsearch.index.mapper.internal.FieldNamesFieldMapper;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class ExistsQueryBuilderTest extends BaseQueryTestCase<ExistsQueryBuilder> {
+
+    @Override
+    protected ExistsQueryBuilder createEmptyQueryBuilder() {
+        return new ExistsQueryBuilder(null);
+    }
+
+    private Collection<String> getFieldNamesFromContext(String fieldName, QueryParseContext context) {
+        if (getCurrentTypes().length > 0 && fieldName.equals(BaseQueryTestCase.OBJECT_FIELD_NAME)) {
+            // "object" field has two inner fields, so we add wildcard create extended field name pattern
+            fieldName = fieldName + ".*";
+        }
+        return context.simpleMatchToIndexNames(fieldName);
+    }
+
+    @Override
+    protected Query createExpectedQuery(ExistsQueryBuilder queryBuilder, QueryParseContext context) throws IOException {
+        final FieldNamesFieldMapper fieldNamesMapper = (FieldNamesFieldMapper)context.mapperService().fullName(FieldNamesFieldMapper.NAME);
+        Collection<String> fields = getFieldNamesFromContext(queryBuilder.name(), context);
+
+        if (fields.isEmpty()) {
+            return Queries.newMatchNoDocsQuery();
+        }
+
+        BooleanQuery boolFilter = new BooleanQuery();
+        for (String field : fields) {
+            if (fieldNamesMapper == null || !fieldNamesMapper.enabled()) {
```

Comment:
shall we flip this outer if? `if (fieldNamesMapper != null && fieldNamesMapper.enabled()) {` looks a bit nicer to me